src/cephadm/cephadm

```diff
@@ -4608,6 +4609,336 @@ def command_install():
+
+    def _get_capacity(self, dev):
+        # type: (str) -> int
+        size_path = os.path.join('/sys/block', dev, 'size')
+        size_blocks = int(rfiles([size_path]))
+        blk_path = os.path.join('/sys/block', dev, 'queue', 'logical_block_size')
+        blk_count = int(rfiles([blk_path]))
+        return size_blocks * blk_count
+
+    def _get_capacity_by_type(self, rota='0'):
+        # type: (str) -> int
+        devs = self._get_devs_by_type(rota=rota)
+        capacity = 0
+        for dev in devs:
+            capacity += self._get_capacity(dev)
+        return capacity
+
+    def _dev_list(self, dev_list):
+        # type: (List[str]) -> List[str]
+        disk_list = list()
+
+        for dev in dev_list:
+            disk_model = rfiles(['/sys/block/{}/device/model'.format(dev)]).strip()
+            vendor = rfiles(['/sys/block/{}/device/vendor'.format(dev)]).strip()
+            disk_vendor = HostFacts._disk_vendor_workarounds.get(vendor, vendor)
+            disk_size_bytes = self._get_capacity(dev)
+            disk_list.append("{} {} ({})".format(disk_vendor, disk_model, bytes_to_human(disk_size_bytes)))
+        return disk_list
+
+    @property
+    def hdd_list(self):
+        # type: () -> List[str]
+        devs = self._get_devs_by_type(rota='1')
+        return self._dev_list(devs)
+
+    @property
+    def flash_list(self):
+        # type: () -> List[str]
+        devs = self._get_devs_by_type(rota='0')
+        return self._dev_list(devs)
+
+    @property
+    def hdd_capacity_bytes(self):
+        # type: () -> int
+        return self._get_capacity_by_type(rota='1')
+
+    @property
+    def hdd_capacity(self):
+        # type: () -> str
+        return bytes_to_human(self.hdd_capacity_bytes)
+
+    @property
+    def flash_count(self):
+        # type: () -> int
+        return len(self._get_devs_by_type(rota='0'))
+
+    @property
+    def flash_capacity_bytes(self):
+        # type: () -> int
+        return self._get_capacity_by_type(rota='0')
+
+    @property
+    def flash_capacity(self):
+        # type: () -> str
+        return bytes_to_human(self.flash_capacity_bytes)
+
+    def _process_nics(self):
+        # type: () -> None
+        for nic_path in HostFacts._nic_path_list:
+            if not os.path.exists(nic_path):
+                continue
+            for iface in os.listdir(nic_path):
+
+                mtu = rfiles([os.path.join(nic_path, iface, 'mtu')])
+                operstate = rfiles([os.path.join(nic_path, iface, 'operstate')])
+                try:
+                    speed = rfiles([os.path.join(nic_path, iface, 'speed')])
+                except OSError:
+                    # OSError is raised when the device doesn't support
+                    # the ethtool get_link_ksettings, so we'll use -1
+                    # to indicate unknown (-1 is also shown for NICs
+                    # that are not link-up)
+                    speed = "-1"
```

Comment:
can we convert this to a number instead of a string?

Reply:
yup.